addons/purchase/models/stock.py

```diff
@@ -60,7 +60,7 @@ def get_price_unit(self):
     def copy(self, default=None):
         self.ensure_one()
         default = default or {}
-        if not default.get('split_from'):
+        if not default.get('split_from') and not default.get('origin_returned_move_id'):
```

Comment:
when do partial receive or refund , and create backorder, at that time
 copy method is called and to link that refund move we have to keep
 purchase line id instead of set it false..